app/controllers/videos_controller.rb

```diff
@@ -1,5 +1,8 @@
 class VideosController < ApplicationController
+  before_action :require_user, except: [:index]
```

Comment:
The require_user method shows a notice on the front-page saying "Only restricted to registered users" so i am bypassing that.